core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1267,6 +1314,24 @@ object KafkaConfig {
       .define(GroupInitialRebalanceDelayMsProp, INT, Defaults.GroupInitialRebalanceDelayMs, MEDIUM, GroupInitialRebalanceDelayMsDoc)
       .define(GroupMaxSizeProp, INT, Defaults.GroupMaxSize, atLeast(1), MEDIUM, GroupMaxSizeDoc)
 
+      /** New group coordinator configs */
+      // All properties are kept internal until KIP-848 is released.
+      // This property meant to be here only during the development of KIP-848. It will
+      // be replaced by a metadata version before releasing it.
+      .defineInternal(NewGroupCoordinatorEnableProp, BOOLEAN, Defaults.NewGroupCoordinatorEnable, null, MEDIUM, NewGroupCoordinatorEnableDoc)
+      .defineInternal(GroupCoordinatorNumThreadsProp, INT, Defaults.GroupCoordinatorNumThreads, atLeast(1), MEDIUM, GroupCoordinatorNumThreadsDoc)
+
+      /** Consumer groups configs */
+      // All properties are kept internal until KIP-848 is released.
+      .defineInternal(ConsumerGroupSessionTimeoutMsProp, INT, Defaults.ConsumerGroupSessionTimeoutMs, atLeast(1), MEDIUM, ConsumerGroupSessionTimeoutMsDoc)
```

Comment:
how does the operator use ConsumerGroupSessionTimeoutMs and ConsumerGroupSessionMin/MaxTimeoutMs configs?

is the idea to set min/max once, then configure the base timeout which the min/max would enforce?

Reply:
They will be mainly used to ensure that the dynamic group configurations are within the defined boundaries.